src/dbnode/storage/index/block.go

```diff
@@ -401,12 +418,30 @@ func (b *block) addCompactedSegmentFromSegments(
 	return append(result, newReadableSeg(compacted, b.opts))
 }
 
+func (b *block) acceptWritesWithRLock() bool {
```

Comment:
This sounds like a set function to accept writes. Maybe `writesAcceptedWithRLock`?